src/dns/coredns.rs

```diff
@@ -133,15 +133,26 @@ impl CoreDns {
                                 }
                             };
                             let mut resolved_ip_list: Vec<IpAddr> = Vec::new();
+                            let mut nameservers_scoped: Vec<ScopedIp> = Vec::new();
+                            // Add resolvers configured for container
                             if let Some(Some(dns_servers)) = self.backend.ctr_dns_server.get(&src_address.ip()) {
                                     if !dns_servers.is_empty() {
-                                        let mut nameservers_scoped: Vec<ScopedIp> = Vec::new();
                                         for dns_server in dns_servers.iter() {
                                             nameservers_scoped.push(ScopedIp::from(*dns_server));
                                         }
+                                    }
+                            }
+                            // Add network scoped resolvers
+                            if let Some(network_dns_servers) = self.backend.get_network_scoped_resolvers(&src_address.ip()) {
```

Comment:
This should not be done if there were container-configured resolvers. We want container-provided resolvers, or network-provided resolvers. Not both.